jsettlers.graphics/src/main/java/jsettlers/graphics/map/controls/original/panel/content/settlers/profession/ProfessionPanel.java

```diff
@@ -1,66 +1,245 @@
+import jsettlers.graphics.ui.UIElement;
 import jsettlers.graphics.ui.UIPanel;
-import jsettlers.graphics.ui.layout.ProfessionsLayout;
 
 public class ProfessionPanel extends AbstractContentProvider implements UiContentUpdater.IUiContentReceiver<IPartitionSettings> {
 
-	private UIPanel panel;
-	private final UiLocationDependingContentUpdater<IPartitionSettings> uiContentUpdater = new UiLocationDependingContentUpdater<>(this::currentDistributionSettingsProvider);
+	private final ContentPanel panel;
+	private final UiLocationDependingContentUpdater<IPartitionSettings> uiContentUpdater;
+	private long lastChangeTimestamp;
 
 	public ProfessionPanel() {
-		panel = new ProfessionsLayout()._root;
-		uiContentUpdater.addListener(this);
+		this.panel = new ContentPanel();
+		this.uiContentUpdater = new UiLocationDependingContentUpdater<>(ProfessionPanel::currentDistributionSettingsProvider);
+		this.uiContentUpdater.addListener(this);
+		this.lastChangeTimestamp = 0;
 	}
 
 	@Override
 	public UIPanel getPanel() {
 		return panel;
 	}
 
-	private IPartitionSettings currentDistributionSettingsProvider(IGraphicsGrid grid, ShortPoint2D position) {
-		if (grid.getPlayerAt(position.x, position.y) != null) {
-			return grid.getPartitionData(position.x, position.y).getPartitionSettings();
-		}
+	@Override
+	public ESecondaryTabType getTabs() {
+		return ESecondaryTabType.SETTLERS;
+	}
 
-		return null;
+	@Override
+	public void showMapPosition(ShortPoint2D position, IGraphicsGrid grid) {
+		this.panel.setPosition(position);
+		super.showMapPosition(position, grid);
+		this.uiContentUpdater.updatePosition(grid, position);
 	}
 
 	@Override
-	public ESecondaryTabType getTabs() {
-		return ESecondaryTabType.SETTLERS;
+	public void contentShowing(ActionFireable actionFireable) {
+		super.contentShowing(actionFireable);
+		this.uiContentUpdater.start();
 	}
 
 	@Override
-	public void update(IPartitionSettings data) {
+	public void update(IPartitionSettings partitionSettings) {
+		// Use delay to prevent race condition between direct ui update and settings update
+		if (System.currentTimeMillis() - lastChangeTimestamp > 5000) {
+			IProfessionSettings professionSettings = partitionSettings.getProfessionSettings();
+			this.panel.setup(professionSettings);
+		}
+	}
+	
+	public void setPlayer(IInGamePlayer player) {
+		uiContentUpdater.setPlayer(player);
+	}
 
+	private static IPartitionSettings currentDistributionSettingsProvider(IGraphicsGrid grid, ShortPoint2D position) {
+		return grid.getPlayerAt(position.x, position.y).getPlayerId() >= 0 ? grid.getPartitionData(position.x, position.y).getPartitionSettings() : null;
 	}
 
-	public enum ConfigurableRatioType {
-		BEARER,
-		DIGGER,
-		BRICKLAYER
+	public enum EProfessionType {
+		CARRIER("Carriers", true, EMovableType.BEARER),
+		DIGGER("Diggers", false, EMovableType.DIGGER),
+		BUILDER("Builders", false, EMovableType.BRICKLAYER);
+
+		public final String label;
```

Comment:
This is only meant as text in the ui, isn't it?
You can just call Labels#getLabel(EMovableType) on EProfessionType#moveableType to get translated labels

Reply:
Thanks for the tip. I implemented it with the Labels#getName(EMoveableType) method.